### PR TITLE
[2290] Request details Approve & Deny button group

### LINKED ISSUE #2290: Add "Approve/Deny Request" button on the Request Details page to be consistent with Privacy Requests page
### Description
Add the ability to Approve or Deny a privacy request from the Request Details page. This should have the exact same behavior as the Approve or Deny options that are accessible from the Privacy Requests page (the table view), but just be accessible from the page for convenience.

#### Current "Reprocess" Button
Today, the only action accessible from the Request Details page is the "Reprocess" button is shown when an error occurs:
Image

#### Approve/Deny Design
For consistency, we should show the same "Approve/Deny" toggle that is visible on the Privacy Requests page today:
Image

This Approve/Deny option should be shown where the "Reprocess" button appears (see above).
 

### Acceptance Criteria
1. I can approve or deny a DSR request from the Request Details
2. Approve behaves exactly the same as if clicked from the Privacy Requests page (e.g. shows a confirmation modal)
2. Deny behaves exactly the same as if clicked from the Privacy Requests page (e.g. shows a confirmation modal with a required field for denial reason)

### PR DESCRIPTION
**Pull request by ssangervasi**
Closes #2290

### Code Changes

- Request details Approve & Deny button group
  - ApproveButton component
  - DenyButton component
  - Tests for approve & deny buttons

- General improvements:
  - StyledButton for shared button styles
  - Use "on" for props and "handle" for handler functions
  - Fix `returnFocusOnClose`
  - Initial tests for requests table and details pages
  - getByTestIdPrefix Cypress command

### Steps to Confirm

* [ ] Details page buttons
  * Approve a request
  * Deny a request with reason modal
* [ ] Table menu buttons
  * Should be verified as well because the components were refactored
  * Approve a request
  * Deny a request with reason modal

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

_Updated video_ after resolving https://github.com/ethyca/fides/pull/2443 merge conflicts:
<video src="https://user-images.githubusercontent.com/2236777/216207332-394757b4-500d-4f8a-b0d7-5512edbb0a94.mp4" />

With the buttons refactored out, they can be dropped right into the page:
details-button

Video of using Deny button & modal:
<video src="https://user-images.githubusercontent.com/2236777/216171028-cdbc5fb4-4b6d-457c-b0d4-5907301f2070.mp4" />

### CODE REVIEW

**Comment by ssangervasi** [comment]
Still resolving the merge conflicts from https://github.com/ethyca/fides/pull/2443 which adds a confirmation for the approve button.

File: clients/admin-ui/src/features/privacy-requests/hooks/useMutations.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+} from "~/features/privacy-requests/privacy-requests.slice";
+import { PrivacyRequestEntity } from "~/features/privacy-requests/types";
+
+export const useMutations = ({
+  subjectRequest,
+}: {
+  subjectRequest: PrivacyRequestEntity;
+}) => {
+  // The `fixedCacheKey` options allows multiple components to reference the same mutation state for
+  // a subject request.
```
Comment (ssangervasi) [inline clients/admin-ui/src/features/privacy-requests/hooks/useMutations.ts:13]: The privacy requests table could benefit from even more shared state management -- there are a lot of callbacks passed deep down the tree. This was a handy way to start: https://redux-toolkit.js.org/rtk-query/usage/mutations#shared-mutation-results

File: clients/admin-ui/src/features/common/ConfirmationModal.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ const ConfirmationModal = ({
     isOpen={isOpen}
     onClose={onClose}
     size="lg"
-    returnFocusOnClose={returnFocusOnClose || true}
+    returnFocusOnClose={returnFocusOnClose ?? true}
```
Comment (ssangervasi) [inline clients/admin-ui/src/features/common/ConfirmationModal.tsx:47]: @TheAndrewJackson - This was always resulting in `true` which made the buttons pop up after the modal closed.

File: clients/admin-ui/src/features/privacy-requests/RequestRow.tsx [hunk 3]
```diff
@@ -45,10 +42,7 @@ const useRequestRow = (request: PrivacyRequestEntity) => {
   const [hovered, setHovered] = useState(false);
   const [focused, setFocused] = useState(false);
   const [menuOpen, setMenuOpen] = useState(false);
-  const [denyModalOpen, setDenyModalOpen] = useState(false);
-  const [approveModalOpen, setApproveModalOpen] = useState(false);
-  const [approveRequest, approveRequestResult] = useApproveRequestMutation();
-  const [denyRequest, denyRequestResult] = useDenyRequestMutation();
+  const [denyRequest] = useDenyRequestMutation();
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/privacy-requests/RequestRow.tsx:45]: Is this still being used in this component? It looks like `denyRequest` may not be called. I think this is handled in the `useMutation` hook now.

File: clients/admin-ui/src/features/privacy-requests/hooks/useMutations.ts (hunk 1 above)
Comment (TheAndrewJackson) [reply clients/admin-ui/src/features/privacy-requests/hooks/useMutations.ts:13]: This is really cool. I didn't know you could do this with RTK

**Review by TheAndrewJackson — COMMENTED** [review]
Thanks for refactoring this and adding the tests! Just one small nit.  Otherwise everything looks great.

In my testing I noticed that the approval modal was styled slightly different between the table and details pages. Did you see this?

table:

Screenshot 2023-02-02 at 16 09 29


details:

Screenshot 2023-02-02 at 16 09 49

**Comment by ssangervasi** [comment]
> In my testing I noticed that the approval modal was styled slightly different between the table and details pages. Did you see this?

Looks like applying the size to the `ButtonGroup` made the style leak through to the modal buttons - I guess because the react context transcends portals 🌌.

**Review by TheAndrewJackson — APPROVED** [review]
I checked again and the styles are working now 👍 Thanks for the edits! :shipit:

**Pull request merged**